expression/builtin_string_test.go

```diff
@@ -349,6 +353,10 @@ func (s *testEvaluatorSuite) TestRight(c *C) {
 		c.Assert(retType.Collate, Equals, t.retType.Collate)
```

Comment:
@SteveZhangBit 
we can move these test cases which are used for testing type inferring
to plan/typeinfer_test.go which is more readable.

same as `space`